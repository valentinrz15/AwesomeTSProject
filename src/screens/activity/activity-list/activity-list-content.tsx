import React, { FunctionComponent } from 'react';
import { View, Text, TouchableOpacity } from 'react-native';
import activityListStyles from './activity-list-styles';

interface ActivityListContentProps {
  data: {
    title: string;
    id: string;
  };
}

export const ActivityListContent: FunctionComponent<
  ActivityListContentProps
> = ({ data }) => (
  <View
    style={activityListStyles.contentList}
    testID="activity-activity-list-content-container">
    <TouchableOpacity
      testID="activity-activity-list-content-button"
      style={activityListStyles.contentButtonList}>
      <Text>{data.title}</Text>
    </TouchableOpacity>
  </View>
);
