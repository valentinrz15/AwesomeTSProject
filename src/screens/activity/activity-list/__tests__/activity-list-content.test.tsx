import React from 'react';
import renderer from 'react-test-renderer';
import { ActivityListContent } from '../activity-list-content';

describe('activity list content test', () => {
  it('should render activity list content snapshot', () => {
    const { toJSON, root } = renderer.create(
      <ActivityListContent data={{ id: '1', title: 'Hola' }} />,
    );
    const { data } = root.props;

    expect(data.id).toBe('1');
    expect(data.title).toBe('Hola');

    expect(toJSON()).toMatchSnapshot();
  });
});
