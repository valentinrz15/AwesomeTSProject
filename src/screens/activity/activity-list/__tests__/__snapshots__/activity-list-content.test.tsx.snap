// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`activity list content test should render activity list content snapshot 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
  testID="activity-activity-list-content-container"
>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderRadius": 10,
        "borderWidth": 1,
        "marginHorizontal": 5,
        "marginVertical": 10,
        "opacity": 1,
        "padding": 5,
      }
    }
    testID="activity-activity-list-content-button"
  >
    <Text>
      Hola
    </Text>
  </View>
</View>
`;
